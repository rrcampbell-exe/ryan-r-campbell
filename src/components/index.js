import BlogTags from './Blog/BlogTags'
import BlogTeaser from './Blog/BlogTeaser'
import Books from './Books/Books'
import Builds from './Builds/Builds'
import Contact from './Contact/Contact'
import Footer from './Footer/Footer'
import Header from './Header/Header'
import PageWrapper from './PageWrapper/PageWrapper'
import SubNav from './SubNav/SubNav'
import Reviews from './Reviews/Reviews'
import NotFound from './NotFound/NotFound'
import NotificationBanner from './NotificationBanner/NotificationBanner'
import Loading from './Loading/Loading'
import Quote from './Quote/Quote'
import TextAndImage from './TextAndImage/TextAndImage'
import HamburgerMenu from './HamburgerMenu/HamburgerMenu'
import PopText from './PopText/PopText'
import PodcastTeaser from './Podcast/PodcastTeaser'
import Boasts from './Tech/Boasts'
import SkillsCloud from './Tech/SkillsCloud'
import BlogFooter from './Blog/BlogFooter'
import SubstackRedirect from './SubstackRedirect/SubstackRedirect'
import { CodeSpan, CodeBlock } from './CodeBlock/CodeBlock'

export { BlogTeaser, Books, Builds, Footer, Header, PageWrapper, SubNav, Contact, Reviews, NotFound, NotificationBanner, Loading, Quote, TextAndImage, BlogTags, HamburgerMenu, PopText, PodcastTeaser, Boasts, SkillsCloud, BlogFooter, CodeSpan, CodeBlock, SubstackRedirect }
