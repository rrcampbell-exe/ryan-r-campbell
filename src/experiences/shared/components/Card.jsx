import React from 'react'
import styled from 'styled-components'

// TODO: size should flex based on number of cards
// TODO: need to set up theme colors

const StyledDiv = styled.div`
  height: 16rem;
  width: 16rem;
  border: 1px solid #242424;
  border-radius: 4px;
  padding: 1rem;
  margin: 1rem;
`

const Card = ({ children }) => {
  return (
    <StyledDiv>
      {children}
    </StyledDiv>
  )
}

export default Card
