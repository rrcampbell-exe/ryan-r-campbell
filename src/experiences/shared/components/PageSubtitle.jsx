import React from 'react'
import styled from 'styled-components'

const StyledH2 = styled.h2`
  margin: 0 auto;
  padding: 0;
`

const PageSubtitle = ({ subtitle }) => {
  return (
    <StyledH2>
      {subtitle}
    </StyledH2>
  )
}

export default PageSubtitle
