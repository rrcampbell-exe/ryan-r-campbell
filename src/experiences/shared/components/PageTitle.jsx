import React from 'react'
import styled from 'styled-components'

const StyledH1 = styled.h1`
  margin: 0 auto;
  padding: 0;
`

const PageTitle = ({ title }) => {
  return (
    <StyledH1>
      {title}
    </StyledH1>
  )
}

export default PageTitle
