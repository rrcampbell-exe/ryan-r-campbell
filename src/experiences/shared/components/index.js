import BlogPost from './Blog/BlogPost'
import BlogTeaser from './Blog/BlogTeaser'
import Books from './Books/Books'
import Builds from './Builds/Builds'
import Footer from './Footer/Footer'
import Header from './Header/Header'
import PageWrapper from './PageWrapper/PageWrapper'
import SubNav from './SubNav/SubNav'
import Contact from './Contact/Contact'
import Reviews from './Reviews/Reviews'
import NotFound from './NotFound/NotFound'
import NotificationBanner from './NotificationBanner/NotificationBanner'

export { BlogPost, BlogTeaser, Books, Builds, Footer, Header, PageWrapper, SubNav, Contact, Reviews, NotFound, NotificationBanner }
