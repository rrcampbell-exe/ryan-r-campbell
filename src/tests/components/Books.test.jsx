import { describe, expect, test } from 'vitest'
import { act, render } from '@testing-library/react'
import ContextAndRouterProvider from '../test-utils'
import { Books } from '../../components'

describe('Books', () => {
  test('should render Books', () => {
    const { getByTestId } = render(
      <ContextAndRouterProvider>
        <Books />
      </ContextAndRouterProvider>
    )

    const books = getByTestId('books')
    expect(books).toBeInTheDocument()
  })
  test('Books default snapshot', () => {
    const { asFragment } = render(
      <ContextAndRouterProvider>
        <Books />
      </ContextAndRouterProvider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
  test('should navigate to /books when a book is clicked', () => {
    const { getAllByTestId } = render(
      <ContextAndRouterProvider>
        <Books />
      </ContextAndRouterProvider>
    )

    const book = getAllByTestId('book-cover')[0]
    act(() => {
      book.click()
    })
    expect(window.location.pathname).toBe('/books')
  })
})
