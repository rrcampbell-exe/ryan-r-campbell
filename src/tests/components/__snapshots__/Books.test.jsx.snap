// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Books > Books default snapshot 1`] = `
<DocumentFragment>
  <div
    class="Books"
    data-testid="books"
  >
    <a
      href="/books"
    />
    <div>
      <a
        href="/books"
      />
      <div
        class="book-container grow-on-hover"
      >
        <a
          href="/books"
        >
          <img
            alt="Orange cover of the book Scambait by Ryan R. Campbell, which displays the book title and shows the main character sitting at a desk with a computer and his Saint Bernard at his side"
            class="book-cover"
            data-testid="book-cover"
            src="./books/scambait/scambait-flat-cover-international-book-awards-readers-favorite-sticker.png"
          />
          <h3
            class="book-title"
          >
            Scambait
          </h3>
          <p>
            The International Book Awards finalist in Humor and Satire that asks, "Who scams the scammers?"
          </p>
        </a>
        <a
          href="/books"
        >
          read more
        </a>
      </div>
    </div>
    <a
      href="/books"
    />
    <div>
      <a
        href="/books"
      />
      <div
        class="book-container grow-on-hover"
      >
        <a
          href="/books"
        >
          <img
            alt="The black cover of the book And Ampersand by Ryan R. Campbell, which features two large ampersands in gold and white, respectively"
            class="book-cover"
            data-testid="book-cover"
            src="./books/and-ampersand/and-ampersand-flat-cover.png"
          />
          <h3
            class="book-title"
          >
            And Ampersand
          </h3>
          <p>
            A collection of short fiction that begins with endings, ends with beginnings, and elevates feel over form.
          </p>
        </a>
        <a
          href="/books"
        >
          read more
        </a>
      </div>
    </div>
    <a
      href="/books"
    />
    <div>
      <a
        href="/books"
      />
      <div
        class="book-container grow-on-hover"
      >
        <a
          href="/books"
        >
          <img
            alt="The blue cover of the book Imminent Dawn by Ryan R. Campbell, which shows the silhouette of a woman against a backdrop of stars and circuitry"
            class="book-cover"
            data-testid="book-cover"
            src="./books/imminent-dawn/imminent-dawn-flat-cover-international-book-awards-sticker.png"
          />
          <h3
            class="book-title"
          >
            Imminent Dawn
          </h3>
          <p>
            The International Book Awards finalist and first installment in the EMPATHY sci-fi saga.
          </p>
        </a>
        <a
          href="/books"
        >
          read more
        </a>
      </div>
    </div>
    <a
      href="/books"
    />
    <div>
      <a
        href="/books"
      />
      <div
        class="book-container grow-on-hover"
      >
        <a
          href="/books"
        >
          <img
            alt="The golden cover of the book Mourning Dove by Ryan R. Campbell, which features the silhouettes of a woman with a backpack and a cat set against a technothriller-like web of connected dots"
            class="book-cover"
            data-testid="book-cover"
            src="./books/mourning-dove/mourning-dove-flat-cover.png"
          />
          <h3
            class="book-title"
          >
            Mourning Dove
          </h3>
          <p>
            An evocative, sweeping symphony of love, revenge, and desperation in cacophonous times.
          </p>
        </a>
        <a
          href="/books"
        >
          read more
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
